drivers/sensor/sgp40/sgp40.c

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright (c) 2021 Leonard Pollak
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT sensirion_sgp40
+
+#include <device.h>
+#include <drivers/i2c.h>
+#include <kernel.h>
+#include <drivers/sensor.h>
+#include <sys/__assert.h>
+#include <logging/log.h>
+#include <sys/crc.h>
+
+#include <drivers/sensor/sgp40.h>
+#include "sgp40.h"
+
+LOG_MODULE_REGISTER(SGP40, CONFIG_SENSOR_LOG_LEVEL);
+
+/*
+ * CRC parameters were taken from the
+ * "Checksum Calculation" section of the datasheet.
+ */
+static uint8_t sgp40_compute_crc(uint16_t value)
+{
+	uint8_t buf[2] = { value >> 8, value & 0xFF };
```

Comment:
`byteorder.h` API should be used.

Reply:
Updated: I made use of the `byteorder.h` API in the sht4x driver aswell.